python/lsst/pipe/tasks/healSparseMappingProperties.py

```diff
@@ -0,0 +1,490 @@
+                          psf_array=None):
+        """Accumulate values from a row of a visitSummary table.
+
+        Parameters
+        ----------
+        indices : `np.ndarray`
+            Indices of values that should be accumulated.
+        ra : `np.ndarray`
+            Array of right ascension for indices
+        dec : `np.ndarray`
+            Array of declination for indices
+        weights : `float` or `np.ndarray`
+            Weight(s) for indices to be accumulated.
+        scalings : `float` or `np.ndarray`
+            Scaling values to coadd zeropoint.
+        row : `lsst.afw.table.ExposureRecord`
+            Row of a visitSummary ExposureCatalog.
+        psf_array : `np.ndarray`, optional
+            Array of approximate psf values matched to ra/dec.
+        """
+        values = self.compute(row, ra, dec, scalings, psf_array=psf_array)
+        if self.config.do_min:
+            self.min_values[indices] = np.fmin(self.min_values[indices], values)
+        if self.config.do_max:
+            self.max_values[indices] = np.fmax(self.max_values[indices], values)
+        if self.config.do_mean:
+            self.mean_values[indices] += values
+        if self.config.do_weighted_mean:
+            self.weighted_mean_values[indices] += weights*values
+        if self.config.do_sum:
+            self.sum_values[indices] += values
+
+    def finalize_mean_values(self, total_weights, total_inputs):
+        """Finalize the accumulation of the mean and weighted mean.
+
+        Parameters
+        ----------
+        total_weights : `np.ndarray`
+            Total accumulated weights, for each value index.
+        total_inputs : `np.ndarray`
+            Total number of inputs, for each value index.
+        """
+        if self.config.do_mean:
+            use, = np.where(total_inputs > 0)
+            self.mean_values[use] /= total_inputs[use]
+        if self.config.do_weighted_mean:
+            use, = np.where(total_weights > 0.0)
+            self.weighted_mean_values[use] /= total_weights[use]
+
+        # And perform any necessary post-processing
+        self.post_process(total_weights, total_inputs)
+
+    def set_map_values(self, pixels):
+        """Assign accumulated values to the maps.
+
+        Parameters
+        ----------
+        pixels : `np.ndarray`
+            Array of healpix pixels (nest scheme) to set in the map.
+        """
+        if self.config.do_min:
+            self.min_map[pixels] = self.min_values
+        if self.config.do_max:
+            self.max_map[pixels] = self.max_values
+        if self.config.do_mean:
+            self.mean_map[pixels] = self.mean_values
+        if self.config.do_weighted_mean:
+            self.weighted_mean_map[pixels] = self.weighted_mean_values
+        if self.config.do_sum:
+            self.sum_map[pixels] = self.sum_values
+
+    def compute(self, row, ra, dec, scalings, psf_array=None):
```

Comment:
That's an interesting point.  Since `compute` is always and only called by `accumulate_values` I didn't want any additional logic since these values would always be available anyway (except perhaps for `psf_array`).  But after your questions and looking at it, I realize that `compute` should actually be `_compute` (similarly for `post_process`) because it should not be a publicly used method.